src/libraries/System.Net.Sockets/tests/FunctionalTests/TcpClientTest.cs

```diff
@@ -493,6 +493,40 @@ public void Connect_Dual_Success()
             }
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void Connect_MappedV4_Success(bool useConnect)
+        {
+            if (!Socket.OSSupportsIPv6 || !Socket.OSSupportsIPv4)
+            {
+                return;
+            }
+
+            using (var server = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            {
+                // Set up a server socket to which to connect
+                server.Bind(new IPEndPoint(IPAddress.Loopback, 0));
+                server.Listen(1);
+                var endpoint = (IPEndPoint)server.LocalEndPoint;
+
+                string serverAddress = $"::ffff:{endpoint.Address.ToString()}";
```

Comment:
Maybe the theory should be parameterized as well as to whether to use an IPv4 or IPv6 address, to validate that the same ctor works well with both?

Reply:
We have many tests to cover the Connect() so I dropped that and add more variants as you suggested for the constructor.